src/cli/repl/meta_cmd.rs

```diff
@@ -649,8 +669,503 @@ impl<F: LurkField + DeserializeOwned> MetaCmd<F> {
+            "        nil)",
+            "      (lambda () (> (car pair) 10))))",
+            "  :rc 10",
+            "  :descr \"example protocol\")",
+        ],
+        run: |repl, args| {
+            let (name, rest) = repl.store.car_cdr(args)?;
+            let (vars, rest) = repl.store.car_cdr(&rest)?;
+            let (body, props) = repl.store.car_cdr(&rest)?;
+            if !matches!(name.tag(), Tag::Expr(ExprTag::Sym)) {
+                bail!(
+                    "Protocol name must be a symbol. Got {}",
+                    name.fmt_to_string(&repl.store, &repl.state.borrow())
+                )
+            }
+            if !matches!(vars.tag(), Tag::Expr(ExprTag::Nil | ExprTag::Cons)) {
+                bail!(
+                    "Protocol vars must be a list. Got {}",
+                    vars.fmt_to_string(&repl.store, &repl.state.borrow())
+                )
+            }
+
+            let prop_map = repl.get_properties(&props, &["rc", "lang", "descr"])?;
+
+            let get_prop = |key, accepts: fn(&Ptr) -> bool, def: fn(&Repl<F>) -> Ptr| -> Result<Ptr> {
+                match prop_map.get(key) {
+                    Some(val) => {
+                        if accepts(val) {
+                            Ok(*val)
+                        } else {
+                            bail!("Invalid value for {key}")
+                        }
+                    }
+                    None => Ok(def(repl)),
+                }
+            };
+
+            let rc = get_prop(
+                "rc",
+                |rc| matches!(rc.tag(), Tag::Expr(ExprTag::Num)),
+                |repl| repl.store.num_u64(repl.rc.try_into().unwrap())
+            )?;
+
+            // TODO: handle lang properly
+            let lang = get_prop(
+                "lang",
+                |_| true, // accept anything for now
+                |repl| repl.store.intern_nil()
+            )?;
+
+            let descr = get_prop(
+                "descr",
+                |descr| matches!(descr.tag(), Tag::Expr(ExprTag::Str)),
+                |repl| repl.store.intern_string("")
+            )?;
+
+            // the standard format for a protocol as Lurk data
+            let protocol = repl.store.list(vec![vars, body, rc, lang, descr]);
+            let binding = repl.store.cons(name, protocol);
+            repl.env = repl.store.cons(binding, repl.env);
+            Ok(())
+        },
+    };
+}
+
+/// Contains the data needed for proof validation (according to some protocol)
+/// and verification
+#[non_exhaustive]
+#[derive(Serialize, Deserialize)]
+#[serde(bound(serialize = "F: Serialize", deserialize = "F: DeserializeOwned"))]
+enum ProtocolProof<
+    'a,
+    F: CurveCycleEquipped,
+    C: Coprocessor<F> + Serialize + DeserializeOwned,
+    M: MultiFrameTrait<'a, F, C>,
+> where
+    <<E1<F> as Engine>::Scalar as ff::PrimeField>::Repr: Abomonation,
+    <<E2<F> as Engine>::Scalar as ff::PrimeField>::Repr: Abomonation,
+{
+    Nova {
```

Comment:
Is there a good reason not to include `SuperNova` from the start?

Reply:
The REPL can't produce SuperNova proofs yet, so I want to leave that for a full patch